composables/useGraphql.ts

```diff
@@ -1,56 +1,49 @@
 import type { ComputedRef } from 'vue'
-import resolveQueryPath from '@/utils/queryPathResolver'
 
-interface DoFetchParams {
-  variables?: Record<string, unknown>
-}
+type Variables = Record<string, unknown>
 
-type UseGraphqlParams<T> = {
-  queryPrefix?: string
+type UseGraphqlParams = {
   queryName: string
-  clientName?: string | ComputedRef<string>
-  variables?: Record<string, unknown> | ComputedRef<Record<string, unknown>>
+  queryPrefix?: string
+  clientName?: MaybeRef<string>
+  variables?: MaybeRef<Variables>
   disabled?: ComputedRef<boolean>
-  data?: Ref<T | undefined>
-  error?: Ref<unknown>
-  loading?: Ref<boolean>
 }
 
 export default function<T = unknown>({
-  queryPrefix = '',
   queryName,
-  clientName = '',
+  queryPrefix,
+  clientName,
   variables = {},
   disabled = computed(() => false),
-  data = ref<T>(),
-  error = ref(),
-  loading = ref(true),
-}: UseGraphqlParams<T>) {
+}: UseGraphqlParams) {
+  const currentRequestId = ref(0) // simple solution to ensure only the latest query result updates the state.
+  const loading = ref(true)
+  const data = ref()
+
   const { client: clientPrefix } = usePrefix()
   const { $consola } = useNuxtApp()
-
   const prefix = queryPrefix || clientPrefix.value
-  const client = clientName || clientPrefix.value
-
-  async function doFetch({
-    variables: extraVariables = {},
-  }: DoFetchParams = {}) {
-    const query = await resolveQueryPath(prefix, queryName)
+  const queryKey = computed(() => [prefix, queryName, JSON.stringify(cloneRawObject(variables))].join('-'))
+  const enabled = computed(() => !unref(disabled))
 
+  const doFetch = async (variables: Variables) => {
     try {
       loading.value = true
-      const { data: result } = await useAsyncQuery<T>({
-        query: query.default,
-        variables: {
-          ...variables,
-          ...extraVariables,
-        },
-        clientId: isRef(client) ? String(client.value) : client,
+      const requestId = ++currentRequestId.value
```

Comment:
We probably don't need the `useGraphql` composable at all. I prefer using `useQuery` directly from NuxtApollo instead - that composable alone might be sufficient

Reply:
Yep, I like this solution a lot better.